sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -88,23 +96,23 @@ case class CreateViewCommand(
     qe.assertAnalyzed()
     val analyzedPlan = qe.analyzed
 
-    if (tableDesc.schema != Nil && tableDesc.schema.length != analyzedPlan.output.length) {
+    if (userSpecifiedColumns.nonEmpty &&
```

Comment:
then? if `userSpecifiedColumns.isEmpty`, what should we check? And it's wrong to not have this check, users don't need to always specify view columns.